### PR TITLE
Feature/new waas auth

### PR DESCRIPTION
**Pull request by BellringerQuinn**
Docs PR: https://github.com/0xsequence/docs/pull/228

This PR is quite large and intimidating based on the number of lines and files changed; however, a lot of this is due to renaming, project restructuring, and the inclusion of the PlayFab SDK.

Renamings, done for devex improvements, highlights:

- WaaSWallet -> SequenceWallet
- WaaSTransactionQueuer -> SequenceWalletTransactionQueuer
- In general, any class, method, or event that contains WaaS in the name and gives you a compile error has had WaaS removed from its name (with the exception of the few previously mentioned classes)
Examples:
OnWaaSWalletCreated -> OnWalletCreated
WaaSSession -> Session
- All namespaces containing "WaaS" have also been replaced with "Sequence.EmbeddedWallet" - this impacts the "using" statements that are used to import 

Project restructuring involved moving the SDK out of the Assets folder and into the Packages folder, acting as a local package. Tests and related files remain in the Assets folder. This was done for two main reasons:

1. The PlayFab sdk unfortunately does not have a package.json and must be imported manually. This means we cannot efficiently handle dependancy conflicts - if we were to include the PlayFab SDK in our SDK it would cause dependancy compile errors for any devs that already use the PlayFab SDK and it would make our SDK incompatible with any other SDKs that have embedded the PlayFab SDK within them. This restructuring allows us to include the SDK in our project for testing purposes without including it in our SDK
2. This will make uploading our SDK to the Unity Asset Store easier and less error prone as we will be able to use the Asset Store Tools uploader to upload our SDK as a LocalPackage directly from our repo/Unity project

Some of the key files for review are:

**Intents:**
https://github.com/0xsequence/sequence-unity/blob/Feature/newWaaSAuth/Packages/Sequence-Unity/Sequence/SequenceSDK/EmbeddedWallet/DataTypes/ParameterTypes/IntentDataInitiateAuth.cs - IntentDataInitiateAuth

https://github.com/0xsequence/sequence-unity/blob/Feature/newWaaSAuth/Packages/Sequence-Unity/Sequence/SequenceSDK/EmbeddedWallet/DataTypes/ParameterTypes/IntentDataOpenSession.cs - IntentDataOpenSession 

https://github.com/0xsequence/sequence-unity/blob/Feature/newWaaSAuth/Packages/Sequence-Unity/Sequence/SequenceSDK/EmbeddedWallet/DataTypes/ParameterTypes/IntentDataFederateAccount.cs - IntentDataFederateAccount

https://github.com/0xsequence/sequence-unity/blob/Feature/newWaaSAuth/Packages/Sequence-Unity/Sequence/SequenceSDK/EmbeddedWallet/DataTypes/ReturnTypes/IntentResponseAccountFederated.cs - IntentResponseAccountFederated

https://github.com/0xsequence/sequence-unity/blob/Feature/newWaaSAuth/Packages/Sequence-Unity/Sequence/SequenceSDK/EmbeddedWallet/DataTypes/ReturnTypes/Account.cs - Account

https://github.com/0xsequence/sequence-unity/blob/Feature/newWaaSAuth/Packages/Sequence-Unity/Sequence/SequenceSDK/EmbeddedWallet/DataTypes/ReturnTypes/IntentResponseAuthInitiated.cs - IntentResponseAuthInitiated

**Connectors** - these are used to handle the different auth methods and their specific logic (such as assembling the initiateAuth and openSession intents)
https://github.com/0xsequence/sequence-unity/blob/Feature/newWaaSAuth/Packages/Sequence-Unity/Sequence/SequenceSDK/EmbeddedWallet/EmailConnector.cs - EmailConnector

https://github.com/0xsequence/sequence-unity/blob/Feature/newWaaSAuth/Packages/Sequence-Unity/Sequence/SequenceSDK/EmbeddedWallet/GuestConnector.cs - GuestConnector

https://github.com/0xsequence/sequence-unity/blob/Feature/newWaaSAuth/Packages/Sequence-Unity/Sequence/SequenceSDK/EmbeddedWallet/OIDCConnector.cs - OIDCConnector

https://github.com/0xsequence/sequence-unity/blob/Feature/newWaaSAuth/Packages/Sequence-Unity/Sequence/SequenceSDK/EmbeddedWallet/PlayFabConnector.cs- PlayFabConnector

**WaaSLogin/SequenceLogin** - the main logic for WaaS authentication in the Unity SDK
https://github.com/0xsequence/sequence-unity/blob/Feature/newWaaSAuth/Packages/Sequence-Unity/Sequence/SequenceSDK/EmbeddedWallet/IWaaSConnector.cs - IWaaSConnector

https://github.com/0xsequence/sequence-unity/blob/Feature/newWaaSAuth/Packages/Sequence-Unity/Sequence/SequenceSDK/EmbeddedWallet/SequenceLogin.cs - SequenceLogin (renamed from WaaSLogin)

**UI**
Further adding to the size of the PR is the newly added Federated Auth Popup Panel prefab. This panel pops up by default when the `EmailAlreadyInUse` error is received from the WaaS API. Based on the `EnableMultipleAccountsPerEmail` flag in SequenceConfig, the UI will have one of two appearances:

- Not enabled 
EmailAlreadyUsed

- Enabled 
EmailAlreadyUsed-NewAccountsPossible
with create new
NewAccountConfirm
 account confirmation page

**ADRs**
I've also included two ADRs:

1. A brief explanation of why we shifted to the new auth system
2. Why we shifted to the local package development workflow 

Approval of this PR will signify approval of both ADRs

### CODE REVIEW

**Comment by BellringerQuinn** [comment]
Please do not merge until the new WaaS API has been merged into production

**Review by taylanpince — APPROVED** [review]
I did a cursory review of this, it would be a good idea for us to walk through the implementation together next week (or maybe in Toronto in person). Let's move it forward in the meantime. Thank you for the massive amount of work on this!

**Pull request merged**
